src/extensions/default/HealthData/HealthDataManager.js

```diff
@@ -0,0 +1,170 @@
+     */
+    function getHealthData() {
+        var result = new $.Deferred(),
+            oneTimeHealthData = {};
+
+        var userUuid = PreferencesManager.getViewState("UUID");
+
+        if (!userUuid) {
+            userUuid = uuid.v4();
+            PreferencesManager.setViewState("UUID", userUuid);
+        }
+
+        oneTimeHealthData.uuid = userUuid;
+        oneTimeHealthData.snapshotTime = (new Date()).getTime();
+        oneTimeHealthData.os = brackets.platform;
+        oneTimeHealthData.userAgent = navigator.userAgent;
+        oneTimeHealthData.osLanguage = brackets.app.language;
+        oneTimeHealthData.bracketsLanguage = brackets.getLocale();
+        oneTimeHealthData.bracketsVersion = brackets.metadata.version;
+
+        HealthDataUtils.getInstalledExtensions()
+            .done(function (userInstalledExtensions) {
+                oneTimeHealthData.installedExtensions = userInstalledExtensions;
+            })
+            .always(function () {
+                return result.resolve(oneTimeHealthData);
+            });
+
+        return result.promise();
+    }
+
+    /**
+     * Send data to the server
+     */
+    function sendHealthDataToServer() {
+        var result = new $.Deferred();
+        
+        getHealthData().done(function (healthData) {
+            
+            var url = brackets.config.healthDataServerURL,
+                data = JSON.stringify(healthData);
+
+            $.ajax({
+                url: url,
+                type: "POST",
+                data: data,
+                dataType: "text",
+                contentType: "text/plain"
+            })
+                .done(function () {
+                    result.resolve();
+                })
+                .fail(function (jqXHR, status, errorThrown) {
+                    console.error("Error in sending health data. Response : " + jqXHR.responseText + ". Status : " + status + ". Error : " + errorThrown);
+                    result.reject();
+                });
+        })
+            .fail(function () {
+                result.reject();
+            });
+
+        return result.promise();
+    }
+
+    /*
+     * Check if the health data is to be sent to the server. If the user has enabled tracking, health data will be sent once every 24 hours.
+     * Send health data to the server if the period is more than 24 hours.
+     * We are sending the data as soon as the user launches brackets. The data will be sent to the server only after the notification dialog 
+     * for opt-out/in is closed.
+     */
+    function checkHealthDataSend() {
+        var result = new $.Deferred(),
+            isHDTracking = prefs.get("healthDataTracking"),
+            notificationDialogShown = PreferencesManager.getViewState("healthDataNotificationShown");
+        
+        window.clearTimeout(timeoutVar);
+        if (isHDTracking && notificationDialogShown) {
+            var lastTimeSent = PreferencesManager.getViewState("lastTimeSentData"),
```

Comment:
Sorry, I know this name was my suggestion earlier, but @MarcelGerber raised a good point: this prefs key could too easily collide with other stuff.  Could we change to `"lastTimeSentHealthData"`?